lib/web3/filter.js

```diff
@@ -149,6 +150,13 @@ var Filter = function (options, methods, formatter, callback) {
         } else {
             self.filterId = id;
 
+            // check if there are get pending callbacks as a consequence
+            // of calling get() with filterId unassigned.
+            self.getPendingCallbacks.forEach(function (cb){
```

Comment:
I would like to rename `getPendingCallbacks` to `getLogsCallbacks`

Apart from that it looks nice!
Did you tested it?